gtn/hd_span.h

```diff
@@ -1,14 +1,18 @@
 #pragma once
 
+#include <algorithm>
+
+#include "device.h"
 #include "gtn/cuda/cuda.h"
 
-#if defined(CUDA)
+#if defined(_CUDA_)
 #define HDTAG __host__ __device__
 #else
 #define HDTAG
 #endif
 
 namespace gtn {
+
```

Comment:
is it possible to add small documentation on HDSpan class

Reply:
For sure.